incubator/prometheus-postgresql-adapter/templates/deployment.yaml

```diff
@@ -0,0 +1,96 @@
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: {{ include "prometheus-postgresql-adapter.fullname" . }}
+  labels:
+    app: "{{ include "prometheus-postgresql-adapter.name" . }}"
+    chart: "{{ include "prometheus-postgresql-adapter.chart" . }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: "{{ include "prometheus-postgresql-adapter.name" . }}"
+      release: "{{ .Release.Name }}"
+  template:
+    metadata:
+      labels:
+        app: "{{ include "prometheus-postgresql-adapter.name" . }}"
+        chart: "{{ include "prometheus-postgresql-adapter.chart" . }}"
+        release: "{{ .Release.Name }}"
+    spec:
+      {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range .Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end }}
+      {{- end }}
+      containers:
+        - name: "{{ include "prometheus-postgresql-adapter.name" . }}"
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: "{{ .Values.image.pullPolicy }}"
+          args:
+          - -log.level={{ .Values.logLevel }}
+          - -pg.host={{ required "A valid PostgreSQL endpoint .Values.pgHost is required!" .Values.pgHost }}
+          - -web.listen-address=:{{ .Values.webListenAddress }}
+          {{- if .Values.pgDatabase }}
+          - -pg.database={{ .Values.pgDatabase }}
+          {{- end }}
+          {{- if .Values.pgPassword }}
+          - -pg.password={{ .Values.pgPassword }}
+          {{- end }}
+          {{- if .Values.pgPort }}
+          - -pg.port={{ .Values.pgPort }}
+          {{- end }}
+          {{- if .Values.pgSchema }}
+          - -pg.schema={{ .Values.pgSchema }}
+          {{- end }}
+          {{- if .Values.pgTablePrefix }}
+          - -pg.table={{ .Values.pgTablePrefix }}
+          {{- end }}
+          {{- if .Values.pgUser }}
+          - -pg.user={{ .Values.pgUser }}
+          {{- end }}
+          {{- range $key, $value := .Values.extraArgs }}
```

Comment:
@niksajakovljevic Here the user can pass any additional flags that they want. In the Helm chart, I tried to cover the base configuration. Using the `extraArgs` is standard in Helm charts teo make them as generic as possible, and easier to maintain in case flags are added in the future.